src/rill.h

```diff
@@ -117,27 +122,23 @@ const char * rill_store_file(const struct rill_store *store);
 unsigned rill_store_version(const struct rill_store *store);
 rill_ts_t rill_store_ts(const struct rill_store *store);
 size_t rill_store_quant(const struct rill_store *store);
-size_t rill_store_keys(const struct rill_store *store);
-size_t rill_store_vals(const struct rill_store *store);
+size_t rill_store_keys_count(const struct rill_store *store, enum rill_col column);
 size_t rill_store_pairs(const struct rill_store *store);
+size_t rill_store_index_len(const struct rill_store *store, enum rill_col col);
+
 
+struct rill_pairs *rill_store_query_value(
+        struct rill_store *store, rill_val_t val, struct rill_pairs *out);
 struct rill_pairs *rill_store_query_key(
         struct rill_store *store, rill_key_t key, struct rill_pairs *out);
-struct rill_pairs *rill_store_scan_keys(
-        struct rill_store *store,
-        const rill_key_t *keys, size_t len,
-        struct rill_pairs *out);
-struct rill_pairs *rill_store_scan_vals(
-        struct rill_store *store,
-        const rill_val_t *vals, size_t len,
-        struct rill_pairs *out);
 
-size_t rill_store_dump_keys(
-        const struct rill_store *store, rill_val_t *out, size_t cap);
-size_t rill_store_dump_vals(
-        const struct rill_store *store, rill_val_t *out, size_t cap);
 
-struct rill_store_it *rill_store_begin(struct rill_store *store);
+size_t rill_store_keys(
```

Comment:
keys is unfortunately not the right name for what we're extracting. More like distinct values for that column but not sure what's the best one word to represent this.